### PR TITLE
New docs website / Responsive page "sidebar" (and "burger" menu)

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This is the second PR for the finalization of the page "sidebar" (the first one is #840).

This involves adding a responsive behaviour for the "sidebar" (controlled by a "burger" menu) and decouple the scrolling of the sidebar from the scrolling of the content.

_Notice: I am going to have a session with @heatherlarsen in which I'll go through the behaviour with her and see if we need to do some changes to the current implementation._

### :hammer_and_wrench: Detailed description

In this PR I have:
- added event handling logic between “burger” menu in the header, the main “application” controller, and the “sidebar”
- refactored breakpoints mixins to simplify their definitions and make them more easy to use
- added `z-index` scale as CSS tokens
- added links to “top level” routes in the sidebar navigation (only for small viewports)
- updated page “header” to use the updated breakpoint mixins
- added responsive behaviour to the “sidebar” (differentiated between homepage an other pages)
  - show/hide via "burger" menu on small viewports
  - fixed on medium/large viewports
  - for the homepage, always show/hide via "burger" menu
  - allowed the sidebar to scroll independently from the content

**Notice**: the actual styling of the sidebar content is done in #845.

👉 👉 👉 **Previews**:
- homepage: https://hds-website-git-new-docs-website-responsive-sidebar-hashicorp.vercel.app/
- details page: https://hds-website-git-new-docs-website-responsive-sidebar-hashicorp.vercel.app/components/alert/

### 🚧 TODOs

- ~~add a reset for the `showSidebarOnSmallViewport` on route change~~
  - tracked here: https://hashicorp.atlassian.net/browse/HDS-1258

### :link: External links

Jira tickets:
- https://hashicorp.atlassian.net/browse/HDS-934 (main ticket)
- https://hashicorp.atlassian.net/browse/HDS-1253 (behaviour)

***

### 👀 Reviewer's checklist:

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/app/styles/pages/application/footer.scss [hunk 1]
```diff
@@ -54,7 +54,6 @@
       text-decoration: underline;
     }
 
-    // `ember-body-class` sets specific classes on the `<body>` element and we use this to have a full black background even in case of overscrolling
     body.application.index & {
```
Comment (MelSumner) [inline website/app/styles/pages/application/footer.scss:57]: question: I wonder if we should be setting body styles in the footer component?

Comment (didoo) [reply website/app/styles/pages/application/footer.scss:57]: we're not setting a body-style, is a parent selector

File: website/app/components/doc/page/sidebar.hbs [hunk 2]
```diff
@@ -1,9 +1,23 @@
-{{#if this.structuredPageTree}}
-  <aside class="doc-page-sidebar" ...attributes>
-    <div class="doc-page-sidebar__inner-wrapper">
+<aside class={{this.classNames}} ...attributes>
+  <div class="doc-page-sidebar__inner-wrapper">
+    <nav class="doc-page-sidebar__top-routes" aria-label="primary page navigation">
+      <ul>
```
Comment (MelSumner) [inline website/app/components/doc/page/sidebar.hbs:4]:
```suggestion
      <ul role="list">
```

If `list-style` is set to `none`, then the `<ul>` element needs the explicit role of `list` in order for Safari to relay the information correctly to assistive tech.

**Review by MelSumner — APPROVED** [review]
There's an accessibility fix recommended but it could be done in a separate PR.

**Pull request closed without merging**